knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -274,6 +277,7 @@ func (r *ReconcileKnativeKafka) transform(manifest *mf.Manifest, instance *serve
 		configureLegacyEventingKafka(instance.Spec.Channel),
 		configureEventingKafka(instance.Spec),
 		ImageTransform(common.BuildImageOverrideMapFromEnviron(os.Environ(), "KAFKA_IMAGE_"), log),
+		operatorcommon.ConfigMapTransform(instance.Spec.Config, logging.FromContext(context.TODO())),
```

Comment:
It is now possible to configure kafka-broker-config in two ways from the same KnativeKafka config and the new way you just implemented takes precedence:
```spec:
  broker:
    enabled: true
    defaultConfig:
      bootstrapServers: "my-cluster-kafka-bootstrap.kafka:9092"
      numPartitions: "10"
      replicationFactor: "3"
  config:
    kafka-broker-config:
      default.topic.partitions: "10"
      default.topic.replication.factor: "3"
      bootstrap.servers: "my-cluster-kafka-bootstrap.kafka:9092"
```
I think this is alright. Just curious if you've thought about it.

Reply:
Maybe there could be a warning if both broker.defaultConfig and config.kafka-broker-config exist. But not necessary

Reply:
I could try exclude the `kafka-broker-config` CM from that `config` thingy :thinking: 
/cc @pierDipi 

OR :-) We remove the `defaultConfig` bits? But I think those are more "first class", than the CM messy config syntax?

Reply:
The CM syntax is awful, yeah. For example, the channel configuration is much better with the KnativeKafka CR bits:
```
spec:
  channel:
    authSecretName
```
vs.
```
spec:
  config:
    config-kafka:
      eventing-kafka: |
        kafka:
          authSecretName: kafka-cluster
```
I was thinking about configuring as much as possible through the ConfigMap syntax and removing the [legacy spec.channel](https://github.com/openshift-knative/serverless-operator/blob/main/knative-operator/pkg/controller/knativekafka/knativekafka_controller.go#L471) configuration as well. But I'm not so sure now.
Perhaps it makes sense to keep the current "first class" configs and also allow to configure anything else by the ConfigMap syntax. I'd probably keep the current PR as is then.

Reply:
I think this is alright. 
What I'd expect is that `defaultConfig` has priority over the other `config.kafka-broker-config` if they are both set.

Reply:
I'd merge the 2 configurations if they are both set.

Reply:
> I think this is alright.
> What I'd expect is that `defaultConfig` has priority over the other `config.kafka-broker-config` if they are both set.

Then I guess, the new stuff must be registered *before* the `configureEventingKafka(instance.Spec) `

Reply:
@mgencur I updated the PR and registered the ConfigMapTransformer _before_